### DISCUSSION TITLE
Craft CMS "Publish Changes" for Next.js Frontend

### DISCUSSION THREAD

**Discussion by paragonn** [craftcms/cms#13790] [Ideas]
We are using Craft CMS more and more as a CMS for Next.js Frontend web projects. We also use Strapi. There is only one feature in Strapi that I really miss in Craft CMS and that is the "Publish (My Changes)" button. Please see the screenshot below:

<img width="974" alt="Screenshot 2023-10-05 at 2 28 35 PM" src="https://github.com/craftcms/cms/assets/1228734/cb24aa16-f24c-49f3-b026-e12289278215">

Basically, the user makes all of their changes and then clicks this Publish (My Changes) button, which triggers the deployment to the frontend Next.js app. Currently, in Craft CMS we are using webhooks, which trigger a deployment whenever an entry is saved, which potentially leads to lots of deployments being triggered. It would be great to have a "Publish (My Changes)" button that would trigger a deployment when the user is ready to do so.

**Comment by brandonkelly** [comment]
We are planning to add a Content Releases feature in Craft 5.0 which sounds like it would solve this. It will enable authors to stage multiple edits to a Release, and publish them all at one time. You could then create a webhook that triggers a deployment whenever a Release is published.

**Reply by MangoMarcus** [reply]
Hi @brandonkelly , can you confirm if this feature will be shipped with v5.0? The [roadmap](https://craftcms.com/roadmap) targets it as 5.x. Thanks!

**Comment by paragonn** [comment]
Thanks for the update @brandonkelly. That sounds like the perfect solution for this challenge in CMS headless configurations.
